### PR TITLE
feat: Upgrade Redis 3 to 4 for LiveQuery

### LINKED ISSUE #8276: Upgrade Redis 3 to 4
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation

Node Redis dependency should be upgraded form 3 to 4, which includes a breaking change, see [migration guide](https://github.com/redis/node-redis/blob/redis%404.0.0/docs/v3-to-v4.md).

### Feature / Enhancement Description

Upgrade to Redis 4 with Parse Sever 6 release.

### Example Use Case
n/a

### Alternatives / Workarounds
n/a

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

The `RedisCacheAdapter` was updated to Redis 4, but the `RedisPubSub` was not.

Closes: #8276

### Approach

Refactors live query publisher to support Redis 4. First commit includes a test with LiveQueryREDISURL that should fail.

### TODOs before merging

- [x] Add tests

### CODE REVIEW

File: src/LiveQuery/ParseCloudCodePublisher.js [hunk 1]
```diff
@@ -8,7 +8,12 @@ class ParseCloudCodePublisher {
   // config object of the publisher, right now it only contains the redisURL,
   // but we may extend it later.
   constructor(config: any = {}) {
-    this.parsePublisher = ParsePubSub.createPublisher(config);
+    (async () => {
```
Comment (mtrezza) [inline src/LiveQuery/ParseCloudCodePublisher.js]: That's a rare pattern `(async()...)();`, can this be refactored?

Comment (dblythy) [reply src/LiveQuery/ParseCloudCodePublisher.js]: Constructors can't be async - this allows using `await` in constructor

Comment (mtrezza) [reply src/LiveQuery/ParseCloudCodePublisher.js]: I'm not sure we want to encourage a coding pattern of async operations in a constructor. Can we let the constructor only initialize the instance and instead maybe call an async static function that calls the constructor and does the async operations?

```
const pub = await ParseCloudCodePublisher.create(config);
```
and
```
class ParseCloudCodePublisher {
  async create(...) { ... }
}
```

File: spec/ParseLiveQueryRedis.spec.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        liveQuery: {
+          classNames: ['TestObject'],
+          redisURL: 'redis://localhost:6379',
+        },
+        liveQueryServerOptions: {
+          redisURL: 'redis://localhost:6379',
+        },
+      });
+      const subscription = await new Parse.Query('TestObject').subscribe();
+      const [, object] = await Promise.all([
```
Comment (mtrezza) [inline spec/ParseLiveQueryRedis.spec.js]: `[, object]` looks strange

**Comment by dblythy** [comment]
This has been tested on a local version of Parse Server with 8 clusters each with a livequery server, all connected to a single redis instance. Breaking as `ParseServer.startLiveQueryServer` now returns a promise and should be `await`ed

**Review by mtrezza — APPROVED** [review]
Looks good! Is this ready for merge?

**Comment by dblythy** [comment]
Yep!

**Comment by mtrezza** [comment]
Is this a feature or a fix?

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-alpha.12](https://github.com/parse-community/parse-server/releases/tag/6.0.0-alpha.12)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/6.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0](https://github.com/parse-community/parse-server/releases/tag/6.0.0)

**Pull request merged**